### PR TITLE
Broaden lifetimes in Hackernews example

### PR DESCRIPTION
**Pull request by benwis**
Switch the lifetimesi n the hackernews example from 'static to non static, giving a bit more flexibility. Also, because Predrag said so.

### CODE REVIEW

File: trustfall/examples/hackernews/adapter.rs [hunk 1]
```diff
@@ -358,10 +358,12 @@ impl BasicAdapter<'static> for HackerNewsAdapter {
...
+        contexts: ContextIterator<'a, Self::Vertex>,
         _type_name: &str,
         coerce_to_type: &str,
-    ) -> ContextOutcomeIterator<'static, Self::Vertex, bool> {
+    ) -> ContextOutcomeIterator<'a, Self::Vertex, bool> {
         resolve_coercion_using_schema(contexts, &SCHEMA, coerce_to_type)
     }
 }
+
+
```
Comment (obi1kenobi) [inline trustfall/examples/hackernews/adapter.rs:369]:
```suggestion
```
perhaps accidentally added

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Thanks for the fix!

**Comment by obi1kenobi** [comment]
I think this just needs a `cargo fmt` and it's good to merge. It's failing on rustfmt right now.

**Comment by benwis** [comment]
> I think this just needs a `cargo fmt` and it's good to merge. It's failing on rustfmt right now.

Formatted

**Pull request merged**